data/src/main/java/org/akvo/flow/data/datasource/DatabaseDataSource.java

```diff
@@ -0,0 +1,186 @@
+ *
+ * Akvo Flow is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.akvo.flow.data.datasource;
+
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.squareup.sqlbrite.BriteDatabase;
+
+import org.akvo.flow.data.entity.ApiDataPoint;
+import org.akvo.flow.data.entity.ApiQuestionAnswer;
+import org.akvo.flow.data.entity.ApiSurveyInstance;
+import org.akvo.flow.database.Constants;
+import org.akvo.flow.database.RecordColumns;
+import org.akvo.flow.database.ResponseColumns;
+import org.akvo.flow.database.SurveyInstanceColumns;
+import org.akvo.flow.database.SurveyInstanceStatus;
+import org.akvo.flow.database.SyncTimeColumns;
+import org.akvo.flow.database.TransmissionStatus;
+import org.akvo.flow.database.britedb.BriteSurveyDbAdapter;
+
+import java.util.Collections;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import rx.Observable;
+
+public class DatabaseDataSource {
+
+    private final BriteSurveyDbAdapter briteSurveyDbAdapter;
+
+    @Inject
+    public DatabaseDataSource(BriteDatabase db) {
+        this.briteSurveyDbAdapter = new BriteSurveyDbAdapter(db);
+    }
+
+    public Observable<Cursor> getDataPoints(@NonNull Long surveyGroupId, @Nullable Double latitude,
+            @Nullable Double longitude, @Nullable Integer orderBy) {
+        int safeOrderBy;
+        if (orderBy == null) {
+            safeOrderBy = Constants.ORDER_BY_NONE;
+        } else {
+            safeOrderBy = orderBy;
+        }
+        switch (safeOrderBy) {
+            case Constants.ORDER_BY_DISTANCE:
+            case Constants.ORDER_BY_DATE:
+            case Constants.ORDER_BY_STATUS:
+            case Constants.ORDER_BY_NAME:
+                return briteSurveyDbAdapter
+                        .getFilteredSurveyedLocales(surveyGroupId, latitude, longitude,
+                                safeOrderBy);
+            default:
+                return briteSurveyDbAdapter.getSurveyedLocales(surveyGroupId);
+        }
+    }
+
+    public Cursor getSyncedTime(long surveyGroupId) {
+        return briteSurveyDbAdapter.getSyncTime(surveyGroupId);
+    }
+
+    public Observable<List<ApiDataPoint>> syncSurveyedLocales(List<ApiDataPoint> apiDataPoints) {
+        if (apiDataPoints == null) {
+            return Observable.<List<ApiDataPoint>>just(Collections.EMPTY_LIST);
+        }
+        syncSurveyedLocale(apiDataPoints);
+        return Observable.just(apiDataPoints);
+    }
+
+    private void syncSurveyedLocale(List<ApiDataPoint> apiDataPoints) {
```

Comment:
I think we can already start moving away from the name `surveyedLocale` and use `datapoint` instead wherever possible